device/CMakeLists.txt

```diff
@@ -1,15 +1,13 @@
 set(POSITION_INDEPENDENT_CODE ON)
 
 set(FBS_FILE ${PROJECT_SOURCE_DIR}/device/simulation/tt_simulation_device.fbs)
-get_filename_component(FBS_FILE_NAME ${FBS_FILE} NAME)
+get_filename_component(FBS_FILE_NAME ${FBS_FILE} NAME_WLE)
 set(FBS_GENERATED_HEADER "${CMAKE_CURRENT_BINARY_DIR}/${FBS_FILE_NAME}_generated.h")
```

Comment:
This filename is `tt_simulation_device_generated.h`, not `tt_simulation_device.fbs_generated.h`.  So updated the line above to ask for without the last ext (WLE).

Reply:
For reference this cuts down on 148 targets in tt-metal being needlessly re-build/linked.